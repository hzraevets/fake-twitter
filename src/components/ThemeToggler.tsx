import React, { useContext } from "react";
import { Switch } from 'antd';

import { ThemeContext } from 'effects/Theme';

export const ThemeToggler = () => {
  const { theme, toggleTheme } = useContext(ThemeContext);

  return (
    <div
      className="absolute top-2 right-2 cursor-pointer select-none"
      onClick={() => toggleTheme()}
    >
      <Switch checked={theme === 'dark'} />
    </div>
  );
};
