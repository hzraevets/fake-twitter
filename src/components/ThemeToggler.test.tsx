import React from 'react';
import { render, screen } from '@testing-library/react';
// @ts-ignore
import { create } from 'react-test-renderer';

import { ThemeToggler } from './ThemeToggler';

describe('ThemeToggler', () => {
  it('render ThemeToggler component', () => {
    render(<ThemeToggler/>);

    // check theme switch button
    expect(screen.getByRole('switch')).toBeInTheDocument();
  });

  it('render ThemeToggler to snapshot', () => {
    const renderer = create(<ThemeToggler/>);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
