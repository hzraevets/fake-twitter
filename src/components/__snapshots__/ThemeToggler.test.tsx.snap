// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggler render ThemeToggler to snapshot 1`] = `
<div
  className="absolute top-2 right-2 cursor-pointer select-none"
  onClick={[Function]}
>
  <button
    aria-checked={false}
    className="ant-switch css-dev-only-do-not-override-diro6f"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="switch"
    type="button"
  >
    <div
      className="ant-switch-handle"
    />
    <span
      className="ant-switch-inner"
    >
      <span
        className="ant-switch-inner-checked"
      />
      <span
        className="ant-switch-inner-unchecked"
      />
    </span>
  </button>
</div>
`;
